dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart

```diff
@@ -42,20 +43,21 @@ class IoEnricherEventProcessor implements EnricherEventProcessor {
 
   @override
   Future<SentryEvent?> apply(SentryEvent event, Hint hint) async {
-    final contexts = event.contexts;
+    event.contexts = await enrich(event.contexts);
+    return event;
+  }
 
+  @override
+  Future<Contexts> enrich(Contexts contexts) async {
     contexts
-      ..device = await _getDevice(event.contexts.device)
-      ..operatingSystem = _getOperatingSystem(event.contexts.operatingSystem)
-      ..runtimes = _getRuntimes(event.contexts.runtimes)
-      ..app = _getApp(event.contexts.app)
-      ..culture = _getSentryCulture(event.contexts.culture);
+      ..device = await _getDevice(contexts.device)
+      ..operatingSystem = _getOperatingSystem(contexts.operatingSystem)
+      ..runtimes = _getRuntimes(contexts.runtimes)
+      ..app = _getApp(contexts.app)
+      ..culture = _getSentryCulture(contexts.culture);
 
     contexts['dart_context'] = _getDartContext();
-
-    event.contexts = contexts;
-
-    return event;
+    return contexts;
```

Comment:
why do we need to return the contexts again if we modify the contexts reference?

couldn't we just do a `Future<void>`?